tests/cpp/common/test_column_matrix.cc

```diff
@@ -9,49 +9,110 @@ namespace xgboost {
 namespace common {
 
 TEST(DenseColumn, Test) {
-  auto dmat = RandomDataGenerator(100, 10, 0.0).GenerateDMatix();
-  GHistIndexMatrix gmat;
-  gmat.Init(dmat.get(), 256);
-  ColumnMatrix column_matrix;
-  column_matrix.Init(gmat, 0.2);
+  size_t max_num_bins[] = {256, 65536, 65537};
+  for (size_t max_num_bin : max_num_bins) {
+    auto dmat = RandomDataGenerator(100, 10, 0.0).GenerateDMatix();
```

Comment:
You have 100 observations in dmatrix, but set max_bins to value to {256, 65536, 65537}.
I dont think that such case covers what you really want.

Reply:
to be honest it makes sense, as the selection of appropriate type (uint8, uint16, uint32) is done by max_bins value only. I want to have this check very fast :)